src/Microsoft.ApiDesignGuidelines.Analyzers/Core/DoNotHideBaseClassMethods.cs

```diff
@@ -0,0 +1,100 @@
+    /// CA1061: Do not hide base class methods
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotHideBaseClassMethodsAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1061";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.DoNotHideBaseClassMethodsTitle), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.DoNotHideBaseClassMethodsMessage), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.DoNotHideBaseClassMethodsDescription), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+
+        public static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            RuleId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.Design,
+            DiagnosticSeverity.Warning, 
+            isEnabledByDefault: true,
+            description: s_localizableDescription,
+            helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182143.aspx",
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.RegisterSymbolAction(SymbolAnalyzer, SymbolKind.Method);
+        }
+
+        private void SymbolAnalyzer(SymbolAnalysisContext context)
+        {
+            var method = (IMethodSymbol)context.Symbol;
+            if (method.IsOverride)  // Since this is common, check for it now
+            {
+                return;
+            }
+
+            foreach (var hiddenMethod in GetMethodsHiddenByMethod(method, method.ContainingType.BaseType))
+            {
+                var diagnostic = Diagnostic.Create(Rule, context.Symbol.Locations[0], method.ToDisplayString(), hiddenMethod.ToDisplayString());
+                context.ReportDiagnostic(diagnostic);
+            }
+        }
+
+        private IEnumerable<IMethodSymbol> GetMethodsHiddenByMethod(IMethodSymbol method, INamedTypeSymbol baseType)
+        {
+            while (true)
+            {
+                if (baseType?.BaseType == null)
+                {
+                    // There are no other base types to check - we're at the top of the hierarchy
```

Comment:
Probably helpful to add a comment that we are at `System.Object` (base framework type)